lua/telescope/actions/init.lua

```diff
@@ -1108,6 +1108,18 @@ actions.which_key = function(prompt_bufnr, opts)
   end
 end
 
+actions._hide_cursor = function(hide)
+  if not hide then
+    return
+  end
+  state.set_global_key("guicursor", vim.o.guicursor)
```

Comment:
Why write this into the global state? You can just write this into the picker (hidden with a `__`), there is no reason to have this around after the picker was closed. (Thats the reason why global state exist, so we can still have access to current selection or prompt input after the picker was closed)